### PR TITLE
Test execution service

### PR DESCRIPTION
**Pull request by williambrownqqq**
use Thread.sleep(); in test - actually bad practice, but we need wait a bit. Any ideas?

### CODE REVIEW

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        closeable.close();
+    }
+    @Test
+    @DisplayName("Test execute method with interruptions")
+    void execute() throws InterruptedException {
+        WebDriver mockWebDriver = mock(WebDriver.class);
+        Supplier<WebDriver> webDriverSupplier = () -> mockWebDriver;
+        List<Step> steps = new ArrayList<>();
+        steps.add(new Step("action1", "value1"));
+        steps.add(new Step("action2", "value2"));
```
Comment (MazurenkoNick) [inline executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java]: Can be deleted for readability purposes.

**Review by MazurenkoNick — APPROVED** [review]

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    @AfterEach
+    void cleanUp() throws Exception {
+        closeable.close();
+    }
+    @Test
+    @DisplayName("Test execute method with interruptions")
+    void execute() throws InterruptedException {
+        WebDriver mockWebDriver = mock(WebDriver.class);
+        Supplier<WebDriver> webDriverSupplier = () -> mockWebDriver;
```
Comment (Lokankara) [inline executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java:45]: Guava's functional primitives can be replaced with Java API
`import java.util.function.Supplier;`

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.openqa.selenium.WebDriver;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import static org.mockito.Mockito.*;
```
Comment (Lokankara) [inline executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java]: Please, replace with a single class imports

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.openqa.selenium.WebDriver;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
```
Comment (Lokankara) [inline executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java]: Unused import statement

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    private ScenarioExecutor scenarioExecutor;
+
+    ScenarioQueueHolder scenarioQueueHolder;
+    private ExecutionService executionService;
+    private BlockingQueue<Scenario> scenarioQueue;
+
+    private AutoCloseable closeable;
+
+    @BeforeEach
+    void setUp() throws IllegalAccessException {
```
Comment (Lokankara) [inline executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java]: Exception 'java.lang.IllegalAccessException' is never thrown in the method

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        steps.add(new Step("action2", "value2"));
+        Scenario scenario1 = new Scenario("TestScenario1", "example1.com", steps);
+        Scenario scenario2 = new Scenario("TestScenario2", "example2.com", steps);
+
+        scenarioQueue.add(scenario1);
+        scenarioQueue.add(scenario2);
+
+        Thread executionThread = new Thread(() -> executionService.execute(webDriverSupplier));
+        executionThread.start();
+        Thread.sleep(1000);
```
Comment (Lokankara) [inline executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java:54]: You can remove this use of "Thread.sleep()"
Without it, the test is also passed

**Review by Lokankara — APPROVED** [review]

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java (hunk 6 above)
Comment (williambrownqqq) [reply executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java:54]: Thanks, done

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java (hunk 5 above)
Comment (williambrownqqq) [reply executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java]: Thanks, done

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java (hunk 4 above)
Comment (williambrownqqq) [reply executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java]: Thanks, done

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java (hunk 3 above)
Comment (williambrownqqq) [reply executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java]: Thanks, done

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java (hunk 2 above)
Comment (williambrownqqq) [reply executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java:45]: Thanks, done

File: executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java (hunk 1 above)
Comment (williambrownqqq) [reply executor-service/src/test/java/executor/service/facade/execution/ExecutionServiceImplTest.java]: Thanks, done

**Pull request merged**